### PR TITLE
add missing output dataset in reqmgr2

### PR DESCRIPTION
**Pull request by ticoann**
@amaltaro, Alan, thanks finding this bug. Please review. Is there any other record missing in reqmgr2 couchdb or want to add?

Patch is tested in my vm.

### CODE REVIEW

File: src/python/WMCore/ReqMgr/Service/Request.py [hunk 1]
```diff
@@ -414,6 +414,23 @@ def delete(self, request_name):
...
+        Update request_args here if additional information need to be put in couchdb
+        """
+        request_args['RequestWorkflow'] = sanitizeURL("%s/%s/%s/spec" % (request_args["CouchURL"], 
+                                            request_args["CouchWorkloadDBName"], workload.name()))['url']
+            
+        request_args['OutputDatasets'] = []
+        # Add the output datasets if necessary
+        # for some bizarre reason OutpuDatasets is list of lists
+        for ds in workload.listOutputDatasets():
+            request_args['OutputDatasets'].append(ds)
```
Comment (amaltaro) [inline src/python/WMCore/ReqMgr/Service/Request.py]: how about extend() instead of append? that should give us a simple list

**Comment by amaltaro** [comment]
hi Seangchan, since you are asking, want to take a look at this one 
https://github.com/dmwm/WMCore/issues/6299
? :-D

**Comment by ticoann** [comment]
Thanks Alan, Now I remember we had another issue.  I will try to update that.

File: src/python/WMCore/ReqMgr/Service/Request.py (hunk 1 above)
Comment (ticoann) [inline src/python/WMCore/ReqMgr/Service/Request.py]: Thanks Alan, acutally can just do following (I have to check what exactly listOutputDatasets() returns but it seems it returns list of str.
```
request_args['OutputDatasets'] = workload.listOutputDatasets()
```

**Comment by ticoann** [comment]
Hi Alan, I added HardTimeout

**Pull request merged**